### ISSUE TITLE
Enable vscode-kubernetes-tools for VS Code on the Web

### ISSUE THREAD

**Issue by isidorn** [Azure/vscode-kubernetes-tools#1007]
:wave: I'm Isidor, a PM from the VS Code team, and we recently announced VS Code for the Web at https://vscode.dev/, which provides a free, zero-install Microsoft Visual Studio Code experience running entirely in your browser.  You can learn more [here](https://code.visualstudio.com/docs/editor/vscode-web). 

We'd like to ensure kubernetes-tools can run in the web, and we have a guide for enabling extensions for the web [here](https://code.visualstudio.com/api/extension-guides/web-extensions). As a couple of highlights:

- In VS Code for the Web, both the UI and extension host run inside the browser.
- A web extension is structured like a regular extension, but with a different main file: it's defined by the `browser` property
- Access to workspace files needs to go through the VS Code file system API accessible at `vscode.workspace.fs`
- There are currently [three ways](https://code.visualstudio.com/api/extension-guides/web-extensions#test-your-web-extension) to test a web extension

I checked your extension and I see there are quite some  dependencies, so this transition to web might not be possible. However it would be cool if you could at least contribute `grammars` for web, and using `when` clauses disable all other contributions which can not work on the web.

Please let us know if you have questions or feedback on enabling your extension for the web - our team would love to learn from your experiences and collaborate. Thank you!

**Comment by bamurtaugh** [comment]
Hi @itowlson! I was wondering if you might have any thoughts on this? We're happy to discuss if you have any questions or feedback. Thank you!

**Comment by itowlson** [comment]
@bamurtaugh The trouble is that a _lot_ of what we do, we do through the `kubectl` and `helm` binaries, which I assume the browser would be unable to shell out to for security reasons.  I'm really not sure it's worth the effort of going through writing `when` clauses on 90% of our commands and views; what's left would likely be a rather sad little rump that it's not worth most people's time to install.

@squillace has the telemetry on what folks actually use, and I'm rather out of that loop - but if memory serves, most of it was viewing cluster resources and viewing logs, both of which depend on an external binary.  So unless you have a solution for external native executables, I'm reluctant to invest time in this.  But if you do, or if you have standard patterns for this - I'm sure you hit it with language servers too - then let me know and we can discuss further!  Thanks!

**Comment by bamurtaugh** [comment]
Thanks for your response @itowlson! That makes sense.

cc @tanhakabir @aeschli for:

> So unless you have a solution for external native executables, I'm reluctant to invest time in this. But if you do, or if you have standard patterns for this - I'm sure you hit it with language servers too - then let me know

**Comment by squillace** [comment]
Yeah, as a CNCF project, I'd have to say we wouldn't see this as a feature unless you were telling us there was a horrible user situation because of this: any number of vs code extensions in oss will just use binaries. Several I built do, for example. So really this is a leading example of that category.
